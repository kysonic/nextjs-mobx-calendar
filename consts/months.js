export const DAYS = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
export const MONTHS = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
export const DAYS_IN_MONTH = 31;
export const DAYS_IN_WEEK = 7;
