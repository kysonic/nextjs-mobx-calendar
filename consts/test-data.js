export const APP_INITIAL_DATA = {
    navigationStore: {
        items: [
            {
                "id": "home",
                "title": "Home",
                "href": "/"
            }
        ]
    },
    dateStore: {
        ISODate: '2019-01-01T00:00:00.000Z'
    }
};

export const CALENDAR_INITIAL_DATA = {
    "2019-04-10T00:00:00.000Z": [
        {
            "id": "abcde1",
            "title": "Action 1",
            "description": "Description of action 1"
        },
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        }
    ],
    "2019-03-11T00:00:00.000Z": [
        {
            "id": "abcde2",
            "title": "Action 2",
            "description": "Description of action 2"
        },
        {
            "id": "abcde3",
            "title": "Action 4",
            "description": "Description of action 4"
        }
    ]
}
