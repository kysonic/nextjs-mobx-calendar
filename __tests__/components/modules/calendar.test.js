import Calendar from '../../../components/modules/calendar/calendar';
import MonthSwitcher from '../../../components/modules/calendar/month-switcher';
import AppStore from '../../../stores/app';
import CalendarStore from '../../../stores/calendar';
import renderer from 'react-test-renderer';
import 'jest-dom/extend-expect';
import {APP_INITIAL_DATA, CALENDAR_INITIAL_DATA} from '../../../consts/test-data';
import {render, fireEvent, waitForElement, cleanup} from 'react-testing-library';
import {autorun} from 'mobx';

describe("Test Calendar component", () => {
    const appStore = new AppStore(APP_INITIAL_DATA);
    const calendarStore = new CalendarStore({items: CALENDAR_INITIAL_DATA});

    test("Snap testing", () => {
        const tree = renderer
            .create(<Calendar appStore={appStore} calendarStore={calendarStore}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});


describe("UI Testing", () => {
    afterEach(cleanup);

    const appStore = new AppStore(APP_INITIAL_DATA);
    const calendarStore = new CalendarStore({items: CALENDAR_INITIAL_DATA});

    test("Month switcher: Click on arrows", async (done) => {
        const {getByTestId} = render(<MonthSwitcher dateStore={appStore.dateStore}/>);
        expect(getByTestId("month")).toHaveTextContent('January');
        appStore.dateStore.incMonth();

    });
});
