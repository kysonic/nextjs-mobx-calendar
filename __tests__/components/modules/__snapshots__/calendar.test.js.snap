// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calendar component Month switcher: Click on arrows and check calendar table 1`] = `
<td
  class="jsx-3390703719"
  data-testid="table-cell-0-1"
>
  <div
    class="jsx-3877038798 calendar-cell is-today"
  >
    <div
      class="jsx-3877038798 day"
    >
      1
    </div>
    <div
      class="jsx-3877038798 plus"
      data-testid="add-event-2019-01-02T00:00:00.000Z"
      role="button"
    >
      +
    </div>
  </div>
</td>
`;

exports[`Test Calendar component Month switcher: Click on arrows and check calendar table 2`] = `
<td
  class="jsx-3390703719"
  data-testid="table-cell-4-3"
>
  <div
    class="jsx-3877038798 calendar-cell "
  >
    <div
      class="jsx-3877038798 day"
    >
      31
    </div>
    <div
      class="jsx-3877038798 plus"
      data-testid="add-event-2019-02-01T00:00:00.000Z"
      role="button"
    >
      +
    </div>
  </div>
</td>
`;

exports[`Test Calendar component Month switcher: Click on arrows and check calendar table 3`] = `
<td
  class="jsx-3390703719"
  data-testid="table-cell-0-4"
>
  <div
    class="jsx-3877038798 calendar-cell "
  >
    <div
      class="jsx-3877038798 day"
    >
      1
    </div>
    <div
      class="jsx-3877038798 plus"
      data-testid="add-event-2019-02-02T00:00:00.000Z"
      role="button"
    >
      +
    </div>
  </div>
</td>
`;

exports[`Test Calendar component Month switcher: Click on arrows and check calendar table 4`] = `
<td
  class="jsx-3390703719"
  data-testid="table-cell-4-4"
/>
`;
